DEVGUIDE.md

```diff
@@ -0,0 +1,722 @@
+
+
+next we create `locustenv/locusthandler/resources` folder and add there test data file `requests.csv` which has the following:
+```
+"Type","Name","Request Count","Failure Count","Median Response Time","Average Response Time","Min Response Time","Max Response Time","Average Content Size","Requests/s","Failures/s","50%","66%","75%","80%","90%","95%","98%","99%","99.9%","99.99%","99.999%","100%"
+"GET","/",10,0,72,75,66,89,2175,0.26,0.00,73,75,86,87,89,89,89,89,89,89,89,89
+"POST","/",5,5,300,323,288,402,157,0.13,0.13,300,330,330,400,400,400,400,400,400,400,400,400
+"GET","/item",24,0,80,79,67,100,2175,0.63,0.00,81,85,86,86,89,92,100,100,100,100,100,100
+"None","Aggregated",39,5,81,109,66,402,1916,1.03,0.13,81,86,87,89,300,330,400,400,400,400,400,400
+```
+
+Now we can run unit tests from the `locustenv/locusthandler` folder with command 
+
+````
+python -m unittest tests/test_locust.py
+````
+
+and all 3 tests should pass.
+
+
+### Configuring LocustHandler to Oxygen
+
+Let's open the python interpreter by running `python` from the `locustenv` directory and check that we can import the locusthandler:
+
+```
+(locustenv) $ python
+Python 3.7.7 
+Type "help", "copyright", "credits" or "license" for more information.
+>>> import locusthandler.locusthandler
+>>>
+```
+
+running this should not produce any errors, and we can import file `locusthandler.py` from `/locusthandler` folder we created. [Read more about packaging python projects from here.](https://packaging.python.org/glossary/#term-import-package) Next we can exit the python intepreter (CTRL + D) and write following lines to the end of `lib/python3.7/site-packages/oxygen/config.yml`:
+
+```
+locusthandler.locusthandler:
+  handler: LocustHandler
+  keyword: run_locust
+  tags: oxygen-locusthandler
+```
+
+### Install demoapp to run tests against
+
+[Next we we install and run demo-app that we run the locust tests against.](https://github.com/robotframework/WebDemo)
+
+Open up another terminal and run following commands:
+
+
+```
+git clone https://github.com/robotframework/WebDemo.git
+cd WebDemo
+python3 demoapp/server.py
+```
+
+### Running Locust with LocustHandler in Robot test
+
+First we install locust to our locustenv virtualenv:
+
+```
+pip install locust
+```
+
+Then we add `locustfile.py` file to `locustenv/locusthandler` folder which contains the commands for the performance test:
+
+```
+from locust import HttpUser, task, between
+
+class QuickstartUser(HttpUser):
+    wait_time = between(5000, 15000)
+
+    @task
+    def index_page(self):
+        self.client.get("/")
+
+```
+
+
+
+Let's write `test.robot` file to `locustenv/locusthandler` folder which contains test case that runs locust from command line:
+
+```
+*** Settings ***
+Library   oxygen.OxygenLibrary
+Library   OperatingSystem
+
+*** Variables ***
+${STATS_FILE}       ${CURDIR}/../example_stats.csv
+${FAILURE_FILE}     ${CURDIR}/../example_failures.csv
+${HISTORY_FILE}     ${CURDIR}/../example_stats_history.csv
```

Comment:
These variables are never used anywhere, maybe remove them from the example?

Reply:
They are generated from the test so I think it is best to delete them also before running the test incase the developer running the test wants to do something with them

Reply:
Ah, you are completely right